### PR TITLE
Type Identifiers in preview bubble and watch node.

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

This PR is to implement type identifiers for the data in preview bubble and watch node. 

Task: https://jira.autodesk.com/browse/DYN-3416

Screen Shot 2022-06-06 at 9 40 27 AM

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers
@QilongTang @zeusongit

### CODE REVIEW

**Comment by reddyashish** [comment]
Fixing the regressions

**Comment by reddyashish** [comment]
All tests passing now.

**Comment by QilongTang** [comment]
Taking a look now

File: src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs [hunk 1]
```diff
@@ -198,10 +206,25 @@ public IEnumerable<int> Levels
...
+        {
+            get { return valueType; }
+            set { valueType = value; RaisePropertyChanged(nameof(ValueType)); }
+        }
+
         #endregion
 
         public WatchViewModel(Action<string> tagGeometry) : this(null, null, tagGeometry, true) { }
 
+        public WatchViewModel(object obj, string path, Action<string> tagGeometry, bool expanded = false) : this(GetStringFromObject(obj), path, tagGeometry, expanded)
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs:223]: Would you add a comment to this new constructor and its purpose?

File: src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs [hunk 2]
```diff
@@ -214,6 +237,72 @@ public WatchViewModel(string label, string path, Action<string> tagGeometry, boo
...
+        }
+
+        private string GetDisplayType(object obj)
+        {
+            TypeCode typeCode = Type.GetTypeCode(obj.GetType());
+            // returning a customized user friendly string instead of just returning the name of the type 
+            switch (typeCode)
+            {
+                case TypeCode.Boolean:
+                    return "bool";
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs:289]: I wonder if we can define these strings as static readonlys?

**Review by QilongTang — COMMENTED** [review]
A few comments then LGTM

**Review by QilongTang — APPROVED** [review]
LGTM, @zeusongit what do you think?

**Review by zeusongit — APPROVED** [review]

**Comment by reddyashish** [comment]
@mjkkirschner Simulated the watch node with big lists and did not notice any difference. Do you mind if I add that in a followup task once we test it in bug-bash?

**Comment by reddyashish** [comment]
Created a spike to investigate other cases for any performance drop. https://jira.autodesk.com/browse/DYN-5010

**Pull request merged**